### PR TITLE
Add s2geometry

### PR DESCRIPTION
**Pull request by benbovy**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- ~~ If static libraries are linked in, the license of the static library is packaged.~~
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by benbovy** [comment]
@conda-forge/help-c-cpp this is ready for review.

`s2geometry` also provides Python bindings (built using SWIG), but apparently this is not working well with conda/conda-forge: https://github.com/google/s2geometry/issues/126. I've tried building it locally (MacOS) in a conda env but I get the same error.

So I think it's fine to only provide s2geometry's C++ headers and built shared libs for now.

I'd be very happy to add maintainers for this recipe if anyone is interested!

**Comment by benbovy** [comment]
@conda-forge/staged-recipes could you have a look when you have time please? I think it's ready to merge.

Regarding the license, s2geometry currently includes parts of [absl](https://github.com/abseil/abseil-cpp), but since it uses the same license I don't think it's a big issue (they plan to use as a third party dependency starting from the next release).

File: recipes/s2geometry/build.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      -DCMAKE_INSTALL_LIBDIR=lib \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DBUILD_SHARED_LIBS=ON \
+      -DBUILD_EXAMPLES=OFF \
+      -UGTEST_ROOT \
+      $SRC_DIR
+
+### Build
+cmake --build . -- -j${CPU_COUNT}
+
```
Comment (ocefpaf) [inline recipes/s2geometry/build.sh:20]: Is there a make check/test that we can run here?

Comment (benbovy) [reply recipes/s2geometry/build.sh:20]: Yes, s2geometry provides many tests, but it's cmake config currently requires getting and building googletest from source. I'm happy to patch `CMakeLists.txt` to use instead the `gtest` library that is shipped with conda-forge, if needed. I tested it locally (MacOS) and it works well. Or I can add a couple of commands to clone the `gtest` git repository...

Comment (ocefpaf) [reply recipes/s2geometry/build.sh:20]: If that is too much trouble we can leave it for later in the feedstock. Let me know when you are "done."

**Comment by benbovy** [comment]
@ocefpaf - I'm "done".

Tests are running for MacOS and Linux. There's only one test that fails on Linux, and I'm not even sure that it's related to any actual s2geometry logic (it might be related to the test code using a custom encoder), so I think we can ignore it for now. I've tried enabling tests on Windows with no success (not sure s2geometry is properly configured for that).

Thanks for the review!

**Pull request merged**